gpustack/utils/network.py

```diff
@@ -19,83 +20,88 @@ def normalize_route_path(path: str) -> str:
     return path
 
 
-def get_first_non_loopback_ip() -> Tuple[str, str]:
+def get_first_non_loopback_ip() -> str:
     """
-    Get the first non-loopback IPv4 address and the interface name of the machine.
+    Get the first non-loopback IPv4 address of the machine.
 
     Returns:
-        A tuple containing the IPv4 address and the interface name.
+        The IPv4 address as a string.
     """
 
-    target_af = socket.AF_INET
-
-    # First try to get the preferred outbound IP
-    outbound_ip, _ = _get_preferred_outbound_ip(target_af)
-    if outbound_ip:
-        ifname = get_ifname_by_ip(outbound_ip, target_af)
-        return outbound_ip, ifname
-
     # Fallback to scanning all interfaces
-    for ifname, addrs in psutil.net_if_addrs().items():
+    for _, addrs in psutil.net_if_addrs().items():
         for addr in addrs:
-            if addr.family == target_af and not addr.address.startswith(
+            if addr.family == socket.AF_INET and not addr.address.startswith(
                 ("127.", "169.254.")
             ):
-                return addr.address, ifname
+                return addr.address
 
     raise Exception("No non-loopback IPv4 address found.")
 
 
-def _get_preferred_outbound_ip(
-    address_family: Optional[socket.AddressFamily] = None,
-) -> Tuple[Optional[str], Optional[socket.AddressFamily]]:
+def get_non_loopback_ip_ifnames() -> Dict[str, Tuple[str, socket.AddressFamily]]:
```

Comment:
Can you explain why we need the mapping now?

Reply:
Updated